client/mysqlbinlog.cc

```diff
@@ -2394,48 +2394,11 @@ get_one_option(const struct my_option *opt, const char *argument, const char *fi
   case OPT_REWRITE_DB:    // db_from->db_to
   {
     /* See also handling of OPT_REPLICATE_REWRITE_DB in sql/mysqld.cc */
-    const char* ptr;
-    const char* key= argument;  // db-from
-    const char* val;            // db-to
-
-    // Skipp pre-space in key
-    while (*key && my_isspace(&my_charset_latin1, *key))
-      key++;
-
-    // Where val begins
-    if (!(ptr= strstr(key, "->")))
-    {
-      sql_print_error("Bad syntax in rewrite-db: missing '->'\n");
-      return 1;
-    }
-    val= ptr + 2;
-
-    // Skip blanks at the end of key
-    while (ptr > key && my_isspace(&my_charset_latin1, ptr[-1]))
-      ptr--;
-
-    if (ptr == key)
+    if (binlog_filter->add_rewrite_db(argument))
     {
-      sql_print_error("Bad syntax in rewrite-db: empty FROM db\n");
+      sql_print_error("Bad syntax in replicate-rewrite-db");
```

Comment:
Ah, in `mysqlbinlog` the option is just named `rewrite-db` (in `mysqld` it is `replicate-rewrite-db`)

Reply:
Done, thanks !